src/errors.rs

```diff
@@ -35,6 +35,42 @@ pub enum PrimaBridgeError {
     Utf8Error { source: Utf8Error },
 }
 
+impl From<reqwest::Error> for PrimaBridgeError {
+    fn from(error: reqwest::Error) -> Self {
+        Self::HttpError {
+            url: error.url().cloned().unwrap_or_else(default_url),
+            source: error,
+        }
+    }
+}
+
+pub enum PrimaBridgeWithMiddlewareError {
```

Comment:
Why do we need this error? Why cannot `MiddlewareError(reqwest_middleware::Error),` stay in the `PrimaBridgeError`?